libraries/standard/mqtt/src/mqtt_state.c

```diff
@@ -978,13 +974,24 @@ MQTTStatus_t MQTT_UpdateStateAck( MQTTContext_t * pMqttContext,
     MQTTQoS_t qos = MQTTQoS0;
     size_t recordIndex = MQTT_STATE_ARRAY_MAX_COUNT;
     MQTTPubAckInfo_t * records = NULL;
-    MQTTStatus_t status = MQTTBadParameter;
+    MQTTStatus_t status = MQTTBadResponse;
```

Comment:
What is the exact reason for this function to return `BadResponse`?  No path in this function is explicitly setting the `status` value as `BadResponse`.  
If the case for it to return `BadResponse` is when no record ID is found for the packet ID, a better return value would be `MQTTNoRecordFound` as this function by itself is not dealing with the transport interface directly...
If we are sticking with `BadResponse`, at least the `@return` documentation of the API should mention the case when it would return `BadResponse`

Reply:
@leegeth and I had an offline conversation regarding this. This function is only used by the process loop to update states from packets received from the network. If the status here isn't set to bad parameter, it will need to be changed by the process loop before returning. Since this function isn't part of the public API, it seemed fine to change the status here instead and bubble it back to the process loop. I can update the documentation

Reply:
One cannot necessarily call this function that is part of the public header file as non-public (even though we will _try_ to discourage its public visibility through doxygen).  The customer application could still call into it if they discover this function for their need. However, the changes related to that discussion is beyond the scope of this PR, so, I am fine with the `@return` documentation updates for this function